fdb-extensions/src/main/java/com/apple/foundationdb/util/Loggable.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Loggable.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2019 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.util;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+
+/**
+ * Associates loggable information with an object as a map.
```

Comment:
Maybe a little bit more detail here would be good.  Something like:

> Associates loggable information with an object as a map.  Record layer log messages are (relatively) well formed.  They are comprised of a static "title" of the message along with a set of keys and values that provide context about the "title" of the message.  For example a "File not found" log entry may include keys and values like `filename="foo"` and `directory="/bar"`.  This makes the logs easy to generally search for all such errors, later extracting the set of files and directories that couldn't be found.  This interface defines the methods that all objects that wish to provide such logging details must implement.

Reply:
Great comment.